src/app/app.component.ts

```diff
@@ -1,18 +1,31 @@
 import { Component } from '@angular/core';
-import { AngularFirestore} from '@angular/fire/firestore';
+import { AngularFirestore, AngularFirestoreCollection, AngularFirestoreDocument} from '@angular/fire/firestore';
 import { Observable } from 'rxjs';
-import 'firebase/firestore';
+import { map } from 'rxjs/operators';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.scss']
 })
 export class AppComponent {
+  private postsCollection: AngularFirestoreCollection<any>;
+
   title = 'ShutterFeed';
-  items: Observable<any[]>;
+  posts: Observable<any[]>;
+
+  constructor(afs: AngularFirestore) {
+    this.postsCollection = afs.collection<any>('posts');
+    this.posts = this.postsCollection.snapshotChanges().pipe(
+      map(actions => actions.map(a => {
+        const data = a.payload.doc.data();
+        const id = a.payload.doc.id;
+        return { id, ...data };
```

Comment:
a is a bit unclear

Reply:
Oh shit this is actually zombie code lol. I'll remove it ASAP.